Currently, only the [`test_finalize_exits`](https://github.com/omisego/plasma-mvp/blob/48ea0983478a242db29195ac1c1f3be5cda9a107/tests/root_chain/contracts/root_chain/test_root_chain.py#L198) test  fails. 🙄💦 



<details>
<summary>Details</summary>


```sh
$ make test

python -m pytest
========================================================= test session starts ==========================================================
platform darwin -- Python 3.7.2, pytest-4.3.0, py-1.7.0, pluggy-0.8.1
rootdir: /Users/akihito1/src/github.com/ackintosh/plasma-mvp, inifile:
collected 37 items

tests/child_chain/test_block.py ..                                                                                               [  5%]
tests/child_chain/test_child_chain.py ........                                                                                   [ 27%]
tests/child_chain/test_child_chain_integration.py .....                                                                          [ 40%]
tests/child_chain/test_transaction.py .                                                                                          [ 43%]
tests/root_chain/contracts/data_structures/test_priority_queue.py .......                                                        [ 62%]
tests/root_chain/contracts/root_chain/test_root_chain.py .....F                                                                  [ 78%]
tests/utils/test_fixed_merkle.py ........                                                                                        [100%]

=============================================================== FAILURES ===============================================================
_________________________________________________________ test_finalize_exits __________________________________________________________

t = <module 'ethereum.tools.tester' from '/Users/akihito1/src/github.com/ackintosh/plasma-mvp/env/lib/python3.7/site-packages/ethereum-2.3.0-py3.7.egg/ethereum/tools/tester.py'>
u = <module 'ethereum.utils' from '/Users/akihito1/src/github.com/ackintosh/plasma-mvp/env/lib/python3.7/site-packages/ethereum-2.3.0-py3.7.egg/ethereum/utils.py'>
root_chain = <ethereum.tools.tester.ABIContract object at 0x10ba83d30>

    def test_finalize_exits(t, u, root_chain):
        two_weeks = 60 * 60 * 24 * 14
        owner, value_1, key = t.a1, 100, t.k1
        tx1 = Transaction(0, 0, 0, 0, 0, 0,
                          NULL_ADDRESS,
                          owner, value_1, NULL_ADDRESS, 0)
        dep1_blknum = root_chain.getDepositBlock()
        root_chain.deposit(value=value_1, sender=key)
        utxo_pos1 = encode_utxo_id(dep1_blknum, 0, 0)
        exit_bond = root_chain.EXIT_BOND()
        root_chain.startDepositExit(utxo_pos1, NULL_ADDRESS, tx1.amount1, sender=key, value=exit_bond)
        t.chain.head_state.timestamp += two_weeks * 2
        assert root_chain.exits(utxo_pos1) == ['0x' + owner.hex(), NULL_ADDRESS_HEX, 100]
        pre_balance = t.chain.head_state.get_balance(owner)
>       root_chain.finalizeExits(sender=t.k2)

tests/root_chain/contracts/root_chain/test_root_chain.py:212:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
env/lib/python3.7/site-packages/ethereum-2.3.0-py3.7.egg/ethereum/tools/tester.py:103: in kall
    gasprice=kwargs.get('gasprice', GASPRICE)
env/lib/python3.7/site-packages/ethereum-2.3.0-py3.7.egg/ethereum/tools/tester.py:194: in tx
    output = self.direct_tx(transaction)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ethereum.tools.tester.Chain object at 0x10dbb9978>, transaction = <Transaction(78e5)>

    def direct_tx(self, transaction):
        self.last_tx = transaction
        if self.last_sender is not None and privtoaddr(
                self.last_sender) != transaction.sender:
            self.last_sender = None
        success, output = apply_transaction(self.head_state, transaction)
        self.block.transactions.append(transaction)
        if not success:
>           raise TransactionFailed()
E           ethereum.tools.tester.TransactionFailed

env/lib/python3.7/site-packages/ethereum-2.3.0-py3.7.egg/ethereum/tools/tester.py:185: TransactionFailed
-------------------------------------------------------- Captured stdout setup ---------------------------------------------------------
Initializing chain from provided state
INFO:eth.block  Block pre-sealed, 4279013 gas used
INFO:eth.chain  Adding to head head=c6745cf3
Saved 4 address change logs
INFO:eth.chain  Added block 1 (5806fdd2) with 1 txs and 4279013 gas
=========================================================== warnings summary ===========================================================
tests/child_chain/test_child_chain.py::test_apply_deposit
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_no_sig
tests/child_chain/test_child_chain.py::test_send_tx_invalid_sig
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain.py::test_submit_block_no_sig
tests/child_chain/test_child_chain.py::test_submit_block_invalid_sig
tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
  /Users/akihito1/src/github.com/ackintosh/plasma-mvp/plasma/root_chain/deployer.py:124: DeprecationWarning: deploy is deprecated in favor of contract.constructor.transact
    }, args=args)

tests/child_chain/test_child_chain.py::test_apply_deposit
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_no_sig
tests/child_chain/test_child_chain.py::test_send_tx_invalid_sig
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
  /Users/akihito1/src/github.com/ackintosh/plasma-mvp/env/lib/python3.7/site-packages/web3-4.5.0-py3.7.egg/web3/utils/decorators.py:14: DeprecationWarning: transact is deprecated in favor of contract.<functions/events>.<method name>.transact
    return self.method(obj, *args, **kwargs)

tests/child_chain/test_child_chain.py::test_apply_deposit
tests/child_chain/test_child_chain.py::test_apply_deposit
tests/child_chain/test_child_chain.py::test_apply_deposit
tests/child_chain/test_child_chain.py::test_apply_deposit
tests/child_chain/test_child_chain.py::test_apply_deposit
tests/child_chain/test_child_chain.py::test_apply_deposit
tests/child_chain/test_child_chain.py::test_apply_deposit
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_with_sig
tests/child_chain/test_child_chain.py::test_send_tx_no_sig
tests/child_chain/test_child_chain.py::test_send_tx_no_sig
tests/child_chain/test_child_chain.py::test_send_tx_no_sig
tests/child_chain/test_child_chain.py::test_send_tx_no_sig
tests/child_chain/test_child_chain.py::test_send_tx_no_sig
tests/child_chain/test_child_chain.py::test_send_tx_no_sig
tests/child_chain/test_child_chain.py::test_send_tx_invalid_sig
tests/child_chain/test_child_chain.py::test_send_tx_invalid_sig
tests/child_chain/test_child_chain.py::test_send_tx_invalid_sig
tests/child_chain/test_child_chain.py::test_send_tx_invalid_sig
tests/child_chain/test_child_chain.py::test_send_tx_invalid_sig
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_send_tx_double_spend
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain.py::test_submit_block
tests/child_chain/test_child_chain.py::test_submit_block_invalid_sig
tests/child_chain/test_child_chain.py::test_submit_block_invalid_sig
tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_transfer
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
  /Users/akihito1/src/github.com/ackintosh/plasma-mvp/env/lib/python3.7/site-packages/web3-4.5.0-py3.7.egg/web3/utils/decorators.py:14: DeprecationWarning: eventFilter is deprecated in favor of contract.events.<event name>.createFilter
    return self.method(obj, *args, **kwargs)

tests/child_chain/test_child_chain_integration.py::test_deposit
tests/child_chain/test_child_chain_integration.py::test_submit_block
tests/child_chain/test_child_chain_integration.py::test_withdraw_transfer
tests/child_chain/test_child_chain_integration.py::test_withdraw_deposit
  /Users/akihito1/src/github.com/ackintosh/plasma-mvp/env/lib/python3.7/site-packages/web3-4.5.0-py3.7.egg/web3/utils/decorators.py:14: DeprecationWarning: call is deprecated in favor of contract.<functions/events>.<method name>.call
    return self.method(obj, *args, **kwargs)

tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_get_min_empty_should_fail
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert_multiple
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert_multiple
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert_multiple
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert_multiple
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert_out_of_order
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert_out_of_order
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert_out_of_order
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_min
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_min
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_min
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_all
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_all
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_all
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_all
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_all
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_all
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_then_insert
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_then_insert
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_then_insert
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_then_insert
tests/root_chain/contracts/root_chain/test_root_chain.py::test_deposit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_deposit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_deposit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_deposit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_deposit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_finalize_exits
tests/root_chain/contracts/root_chain/test_root_chain.py::test_finalize_exits
tests/root_chain/contracts/root_chain/test_root_chain.py::test_finalize_exits
tests/root_chain/contracts/root_chain/test_root_chain.py::test_finalize_exits
tests/root_chain/contracts/root_chain/test_root_chain.py::test_finalize_exits
tests/root_chain/contracts/root_chain/test_root_chain.py::test_finalize_exits
  /Users/akihito1/src/github.com/ackintosh/plasma-mvp/env/lib/python3.7/site-packages/ethereum-2.3.0-py3.7.egg/ethereum/abi.py:493: DeprecationWarning: encode is deprecated, please use encode_function_call
    DeprecationWarning)

tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert_multiple
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert_multiple
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_insert_out_of_order
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_min
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_min
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_all
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_all
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_all
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_then_insert
tests/root_chain/contracts/data_structures/test_priority_queue.py::test_priority_queue_delete_then_insert
tests/root_chain/contracts/root_chain/test_root_chain.py::test_deposit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_deposit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_deposit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_deposit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_deposit_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_fee_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_start_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_challenge_exit
tests/root_chain/contracts/root_chain/test_root_chain.py::test_finalize_exits
tests/root_chain/contracts/root_chain/test_root_chain.py::test_finalize_exits
tests/root_chain/contracts/root_chain/test_root_chain.py::test_finalize_exits
  /Users/akihito1/src/github.com/ackintosh/plasma-mvp/env/lib/python3.7/site-packages/ethereum-2.3.0-py3.7.egg/ethereum/abi.py:499: DeprecationWarning: decode is deprecated, please use decode_function_result
    DeprecationWarning)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========================================== 1 failed, 36 passed, 294 warnings in 35.74 seconds ==========================================
make: *** [test] Error 1
```

</details>

--- a/.travis.yml
+++ b/.travis.yml
@@ -11,7 +11,7 @@ env:
 install:
   - nvm install $NODE_VERSION
   - npm install --global ganache-cli
-  - wget https://github.com/ethereum/solidity/releases/download/v0.4.24/solc-static-linux && chmod +x ./solc-static-linux && sudo mv solc-static-linux /usr/bin/solc
+  - wget https://github.com/ethereum/solidity/releases/download/v0.5.4/solc-static-linux && chmod +x ./solc-static-linux && sudo mv solc-static-linux /usr/bin/solc
   - pip install --upgrade pip setuptools flake8
   - make
 

--- a/README.md
+++ b/README.md
@@ -32,7 +32,7 @@ Plasma MVP is split into four main parts: `root_chain`, `child_chain`, `client`,
 
 This project has a few pre-installation dependencies.
 
-#### [Solidity](https://solidity.readthedocs.io/en/latest/installing-solidity.html)
+#### [Solidity ^0.5.0](https://solidity.readthedocs.io/en/latest/installing-solidity.html)
 
 Mac:
 ```sh

--- a/tests/root_chain/contracts/root_chain/test_root_chain.py
+++ b/tests/root_chain/contracts/root_chain/test_root_chain.py
@@ -209,7 +209,7 @@ def test_finalize_exits(t, u, root_chain):
     t.chain.head_state.timestamp += two_weeks * 2
     assert root_chain.exits(utxo_pos1) == ['0x' + owner.hex(), NULL_ADDRESS_HEX, 100]
     pre_balance = t.chain.head_state.get_balance(owner)
-    root_chain.finalizeExits(sender=t.k2)
+    root_chain.finalizeExits(NULL_ADDRESS, sender=t.k2)
     post_balance = t.chain.head_state.get_balance(owner)
     assert post_balance == pre_balance + value_1 + exit_bond
     assert root_chain.exits(utxo_pos1) == [NULL_ADDRESS_HEX, NULL_ADDRESS_HEX, value_1]

